processing/src/main/java/io/druid/query/aggregation/NullableAggregatorFactory.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation;
+
+
+import io.druid.common.config.NullHandling;
+import io.druid.segment.ColumnSelectorFactory;
+import io.druid.segment.ColumnValueSelector;
+
+/**
+ * abstract class with functionality to wrap aggregator/bufferAggregator/combiner to make them Nullable.
+ * Implementations of {@link AggregatorFactory} which needs to Support Nullable Aggregations are encouraged
+ * to extend this class.
+ */
+public abstract class NullableAggregatorFactory extends AggregatorFactory
+{
+  @Override
+  public final Aggregator factorize(ColumnSelectorFactory metricFactory)
+  {
+    ColumnValueSelector selector = selector(metricFactory);
+    Aggregator aggregator = factorize(metricFactory, selector);
+    return NullHandling.replaceWithDefault() ? aggregator : new NullableAggregator(aggregator, selector);
+  }
+
+  @Override
+  public final BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory)
+  {
+    ColumnValueSelector selector = selector(metricFactory);
+    BufferAggregator aggregator = factorizeBuffered(metricFactory, selector);
+    return NullHandling.replaceWithDefault() ? aggregator : new NullableBufferAggregator(aggregator, selector);
+  }
+
+  protected abstract BufferAggregator factorizeBuffered(
+      ColumnSelectorFactory metricFactory,
+      ColumnValueSelector selector
+  );
+
+  @Override
+  public final AggregateCombiner makeAggregateCombiner()
+  {
+    AggregateCombiner combiner = makeAggregateCombiner2();
+    return NullHandling.replaceWithDefault() ? combiner : new NullableAggregateCombiner(combiner);
+  }
+
+  @Override
+  public final int getMaxIntermediateSize()
+  {
+    return getMaxIntermediateSize2() + (NullHandling.replaceWithDefault() ? 0 : Byte.BYTES);
+  }
+
+  // ---- ABSTRACT METHODS BELOW ------
```

Comment:
Please add docs to those methods, for extension writers.

Reply:
done.